File: 56-hard-orthogonal-segment-crossings/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type input struct {
	Horizontal [][3]int64 `json:"horizontal"`
	Vertical   [][3]int64 `json:"vertical"`
}

func main() {
	var in input
	if err := json.NewDecoder(os.Stdin).Decode(&in); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	fmt.Println(solve(in.Horizontal, in.Vertical))
}

File: 56-hard-orthogonal-segment-crossings/rust/src/main.rs
use orthogonal_segment_crossings::solve;
use orthogonal_segment_crossings::Input;
use std::io::{self, Read};

fn main() {
    let mut buffer = String::new();
    io::stdin().read_to_string(&mut buffer).unwrap();
    let input: Input = serde_json::from_str(&buffer).unwrap();
    println!("{}", solve(&input.horizontal, &input.vertical));
}
